### ISSUE TITLE
Add another parsing method

### ISSUE THREAD

**Issue by MaurizioRicci** [qbittorrent/search-plugins#53]
Often desired information in a web site, are grouped under class name. For example in some site a list of torrent are a list of div with a particular class. So in some case it would be better to find elements by ID, ClassName or by Type instead of parsing page with the standard html parser and using various flags or variable to remember the state during parsing.

The question is, what about adding another parsing method? Something like jQuery, maybe pyquery or BeautifulSoup: 
https://pythonhosted.org/pyquery/ 
https://www.crummy.com/software/BeautifulSoup/bs4/doc/

**Comment by hannsen** [comment]
What do you propose, that they distribute BeautifulSoup with Qbit?

**Comment by MaurizioRicci** [comment]
@hannsen Yes I was thinking something like that. I don't think that it will require too effort and it may help a lot people. What do you think?

**Comment by ngosang** [comment]
I think beautifulsoup4 is perfect for this, but we have to include the package in the qbittorrent repository so the user don't have to install external packages. I can do it but I think @sledgehammer999  will oppose...

**Comment by MaurizioRicci** [comment]
I understand, never mind. Mine was just a suggestion

**Comment by MaurizioRicci** [comment]
I was thinking about all the site whose content are inside a table. It
would be easy to find all tr and then iterating all td inside each tr.
Instead of parsing whole page with flag. My request was intended as a
suggestion if it could be done easily. I don't want to give you hard work

Maurizio Ricci

Il giorno sab 1 dic 2018, 20:26 Ni Ndogo <notifications@github.com> ha
scritto:

**Comment by nindogo** [comment]
Hi,

I have found for many of those, the re module can usually help.

**Comment by hannsen** [comment]
yeah I used a lot regex, too. It's faster than parsing but also not very readable
